models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatOptions.java

```diff
@@ -28,6 +28,7 @@
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import com.google.cloud.vertexai.api.SafetySetting;
```

Comment:
Spilling, an underlying, 3rd party code in the Spring AI API, is a design anti-pattern.
But trying to create Spring AI shallow wrappers doesn't make much sense either.
I guess, given that the `VertextAiGeminiChatModelOptions` is meant to allow breaking portability (e.g., offers model-specific configurations), means that we can allow such spilling. Furthermore we've done it with other provides too. I hope it won't bite us later.

On the other hand i'm not sure you can use the `spring.ai.vertex.ai.gemini.chat.options.SafetySettings` to configure the SafetySettings declaratively via properties? have you tried this.

Reply:
here is a claude generated POJO class from the protobuff class that would be friendly for declarative configuration in yaml

```
/**
 * A Plain Old Java Object (POJO) representing safety settings for content filtering.
 */
public class SafetySetting {
    
    /**
     * Enum representing different threshold levels for blocking harmful content.
     */
    public enum HarmBlockThreshold {
        UNSPECIFIED,
        BLOCK_LOW_AND_ABOVE,    // Block low threshold and above (more strict)
        BLOCK_MEDIUM_AND_ABOVE, // Block medium threshold and above
        BLOCK_ONLY_HIGH,        // Block only high threshold (less strict)
        BLOCK_NONE,             // Don't block any content
        OFF                     // Turn off safety filter completely
    }
    
    /**
     * Enum representing methods for evaluating harmful content.
     */
    public enum HarmBlockMethod {
        UNSPECIFIED,
        SEVERITY,    // Uses both probability and severity scores
        PROBABILITY  // Uses only probability score
    }
    
    /**
     * Enum representing different categories of harmful content.
     */
    public enum HarmCategory {
        UNSPECIFIED,
        HATE_SPEECH,
        DANGEROUS_CONTENT,
        HARASSMENT,
        SEXUAL_CONTENT
    }
    
    private HarmCategory category;
    private HarmBlockThreshold threshold;
    private HarmBlockMethod method;
    
    // Default constructor
    public SafetySetting() {
        this.category = HarmCategory.UNSPECIFIED;
        this.threshold = HarmBlockThreshold.UNSPECIFIED;
        this.method = HarmBlockMethod.UNSPECIFIED;
    }
    
    // Constructor with all fields
    public SafetySetting(HarmCategory category, HarmBlockThreshold threshold, HarmBlockMethod method) {
        this.category = category;
        this.threshold = threshold;
        this.method = method;
    }
    
    // Getters and setters
    public HarmCategory getCategory() {
        return category;
    }
    
    public void setCategory(HarmCategory category) {
        this.category = category;
    }
    
    public HarmBlockThreshold getThreshold() {
        return threshold;
    }
    
    public void setThreshold(HarmBlockThreshold threshold) {
        this.threshold = threshold;
    }
    
    public HarmBlockMethod getMethod() {
        return method;
    }
    
    public void setMethod(HarmBlockMethod method) {
        this.method = method;
    }
    
    @Override
    public String toString() {
        return "SafetySetting{" +
               "category=" + category +
               ", threshold=" + threshold +
               ", method=" + method +
               '}';
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        
        SafetySetting that = (SafetySetting) o;
        
        if (category != that.category) return false;
        if (threshold != that.threshold) return false;
        return method == that.method;
    }
    
    @Override
    public int hashCode() {
        int result = category != null ? category.hashCode() : 0;
        result = 31 * result + (threshold != null ? threshold.hashCode() : 0);
        result = 31 * result + (method != null ? method.hashCode() : 0);
        return result;
    }
}
```